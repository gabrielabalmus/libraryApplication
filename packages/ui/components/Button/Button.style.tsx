import Button, { ButtonProps } from "@mui/material/Button";
import { styled as MuiStyled } from "@mui/material/styles";
import collorPalette from "@/theme/colorPalette";

const commonStyle = {
  height: "38px",
  padding: "6px 20px",
  letterSpacing: "0.5px",
  fontSize: "15px",
  minWidth: "170px",
};

export const ContainedButton = MuiStyled(Button)<ButtonProps>(({ theme }) =>
  theme.unstable_sx({
    ...commonStyle,
    backgroundColor: collorPalette.primary.base,
    color: collorPalette.white,
    border: `2px solid ${collorPalette.primary.base}`,
    "&:hover": {
      backgroundColor: collorPalette.primary.base,
    },
  })
) as typeof Button;

export const OutlinedButton = MuiStyled(Button)<ButtonProps>(({ theme }) =>
  theme.unstable_sx({
    ...commonStyle,
    backgroundColor: collorPalette.white,
    color: collorPalette.black,
    border: `2px solid ${collorPalette.primary.base}`,
    "&:hover": {
      backgroundColor: collorPalette.white,
    },
  })
) as typeof Button;
