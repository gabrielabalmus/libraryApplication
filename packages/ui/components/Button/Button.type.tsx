export interface ButtonProps {
  variant: ButtonVariant;
  title: string;
  onClick?: () => void;
  type?: ButtonType;
  disabled?: boolean;
  width?: string;
}

export enum ButtonVariant {
  contained = "contained",
  outlined = "outlined",
}

export enum ButtonType {
  submit = "submit",
  button = "button",
  reset = "reset",
}
