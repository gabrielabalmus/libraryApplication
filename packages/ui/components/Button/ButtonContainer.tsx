import { ContainedButton, OutlinedButton } from "./Button.style";
import { ButtonVariant, ButtonProps } from "./Button.type";

const ButtonContainer: React.FC<ButtonProps> = ({
  title,
  variant,
  handleClick,
}) => {
  if (variant === ButtonVariant.contained) {
    return (
      <ContainedButton disableRipple={true} onClick={handleClick}>
        {title}
      </ContainedButton>
    );
  }

  return (
    <OutlinedButton disableRipple={true} onClick={handleClick}>
      {title}
    </OutlinedButton>
  );
};

export default ButtonContainer;
