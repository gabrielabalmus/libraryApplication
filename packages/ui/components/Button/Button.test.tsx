import { render } from "@testing-library/react";
import Button from ".";
import { ButtonProps, ButtonType, ButtonVariant } from "./Button.type";

const mockCLick = jest.fn();

const renderButton = ({ ...props }: ButtonProps) =>
  render(<Button {...props} onClick={mockCLick} />);

describe("snapshot tests for Button", () => {
  it.each([
    ["Test button1", ButtonVariant.contained, ButtonType.button, false, "100px"],
    ["Test button2", ButtonVariant.outlined, ButtonType.submit, true, "150px"],
    ["Test button3", ButtonVariant.outlined, ButtonType.reset, false, "250px"],
  ])("renders correctly", (title, variant, type, disabled, width) => {
    const { container } = renderButton({
      title,
      variant,
      type,
      disabled,
      width,
    });

    expect(container).toMatchSnapshot();
  });
});
