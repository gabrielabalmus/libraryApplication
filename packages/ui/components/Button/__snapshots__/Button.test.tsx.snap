// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for Button renders correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-gx99rr-MuiButtonBase-root-MuiButton-root"
    style="min-width: 100px;"
    tabindex="0"
    type="button"
  >
    Test button1
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`snapshot tests for Button renders correctly 2`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1k3xh08-MuiButtonBase-root-MuiButton-root"
    disabled=""
    style="min-width: 150px;"
    tabindex="-1"
    type="submit"
  >
    Test button2
  </button>
</div>
`;

exports[`snapshot tests for Button renders correctly 3`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1k3xh08-MuiButtonBase-root-MuiButton-root"
    style="min-width: 250px;"
    tabindex="0"
    type="reset"
  >
    Test button3
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
